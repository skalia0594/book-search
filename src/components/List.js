import React from 'react'
import SortButton from './SortButton'
import Book from './Book'

export default function List({
  isloading,
  matchedRecord,
  disableTitleSort,
  disableYearSort,
  noRecordFound,
  sortTitle,
  sortPublishYear
}) {
  return (
    <div>
      {!isloading ?
        <>
          <SortButton
            matchedRecord={matchedRecord}
            disableTitleSort={disableTitleSort}
            disableYearSort={disableYearSort}
            sortTitle={sortTitle}
            sortPublishYear={sortPublishYear}
          />
          {noRecordFound && <span>No Record Found</span>
          }
          <ul>
            {matchedRecord.length > 0 && matchedRecord.map((book) => {
              const title = book.title
              const authors = book.author_name
              const numberOfAuthors = authors && authors.length
              const bookCover = book.cover_i && `https://covers.openlibrary.org/b/id/${book.cover_i}-M.jpg`
              return <Book 
                        key={book.key}
                        book={book}
                        bookCover={bookCover}
                        title={title}
                        authors={authors}
                        numberOfAuthors={numberOfAuthors}
                      />
            }
            )}
          </ul>
        </>
        :
        <span>loading...</span>
      }        
    </div>
  )
}
