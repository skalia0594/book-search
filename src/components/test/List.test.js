import React from "react"
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import List from '../List'


let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div")
  document.body.appendChild(container)
})

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

it("should render a App", () => {
  const docs= [
    {
      "cover_i": 258027,
      "title": "The Lord of the Rings",
      "author_name": [
          "J. R. R. Tolkien"
      ],
      "first_publish_year": 1954,
      "key": "OL27448W",
    },
  ]
  act(() => {
    render(<List matchedRecord={docs} />, container)
  })
  expect(container.innerHTML).toMatchSnapshot()
})
