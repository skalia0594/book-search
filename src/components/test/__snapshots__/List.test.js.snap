// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a App 1`] = `"<div><div><div class=\\"\\"><label>Sort: </label><button type=\\"button\\" class=\\"button\\" data-message=\\"This is sort by title button\\">Title</button><button type=\\"button\\" class=\\"button\\" data-message=\\"This is sort by recent publish year button\\">Recent Publish Year</button></div></div><ul><li><div class=\\"media\\"><div class=\\"image\\"><img src=\\"https://covers.openlibrary.org/b/id/258027-M.jpg\\" alt=\\"The Lord of the Rings\\" title=\\"The Lord of the Rings\\"></div><div class=\\"content\\"><div class=\\"fields\\"><label for=\\"title\\">Book Title:</label><span name=\\"title\\">The Lord of the Rings</span></div><div class=\\"fields\\"><label for=\\"author\\">Book Author:</label><span name=\\"author\\">J. R. R. Tolkien</span></div><div class=\\"fields\\"><label for=\\"publishYear\\">Published Year:</label><span name=\\"publishYear\\">1954</span></div></div></div></li></ul></div>"`;
